libs/components/datetime/testing/src/date-range-picker/date-range-picker-harness.spec.ts

```diff
@@ -0,0 +1,324 @@
+
+    fixture.componentInstance.labelText = 'label';
+    fixture.componentInstance.helpPopoverContent = 'content';
+    fixture.detectChanges();
+
+    await dateRangePickerHarness.clickHelpInline();
+
+    await expectAsync(
+      dateRangePickerHarness.getHelpPopoverContent(),
+    ).toBeResolved();
+  });
+
+  it('should get help popover content', async () => {
+    const { dateRangePickerHarness, fixture } = await setupTest();
+
+    fixture.componentInstance.labelText = 'label';
+    fixture.componentInstance.helpPopoverContent = 'content';
+    fixture.detectChanges();
+
+    await dateRangePickerHarness.clickHelpInline();
+
+    await expectAsync(
+      dateRangePickerHarness.getHelpPopoverContent(),
+    ).toBeResolvedTo(fixture.componentInstance.helpPopoverContent);
+  });
+
+  it('should get help popover title', async () => {
+    const { dateRangePickerHarness, fixture } = await setupTest();
+
+    fixture.componentInstance.labelText = 'label';
+    fixture.componentInstance.helpPopoverTitle = 'title';
+    fixture.componentInstance.helpPopoverContent = 'content';
+    fixture.detectChanges();
+
+    await dateRangePickerHarness.clickHelpInline();
+
+    await expectAsync(
+      dateRangePickerHarness.getHelpPopoverTitle(),
+    ).toBeResolvedTo(fixture.componentInstance.helpPopoverTitle);
+  });
+
+  it('should get hint text', async () => {
+    const { dateRangePickerHarness, fixture } = await setupTest();
+
+    fixture.componentInstance.hintText = 'hint text';
+    fixture.detectChanges();
+
+    await expectAsync(dateRangePickerHarness.getHintText()).toBeResolvedTo(
+      fixture.componentInstance.hintText,
+    );
+  });
+
+  it('should get label', async () => {
+    const { dateRangePickerHarness, fixture } = await setupTest();
+
+    fixture.componentInstance.labelText = 'label';
+    fixture.detectChanges();
+
+    await expectAsync(dateRangePickerHarness.getLabelText()).toBeResolvedTo(
+      fixture.componentInstance.labelText,
+    );
+  });
+
+  it('should get whether date range picker is disabled', async () => {
+    const { dateRangePickerHarness, fixture } = await setupTest();
+
+    fixture.componentInstance.myForm.controls['testPicker'].disable();
+    fixture.detectChanges();
+
+    await expectAsync(dateRangePickerHarness.isDisabled()).toBeResolvedTo(true);
+  });
+
+  it('should get whether date range picker is stacked', async () => {
+    const { dateRangePickerHarness, fixture } = await setupTest();
+
+    fixture.componentInstance.stacked = true;
+    fixture.detectChanges();
+
+    await expectAsync(dateRangePickerHarness.isStacked()).toBeResolvedTo(true);
+  });
+
+  it('should get whether start date required error is thrown', async () => {
+    const { dateRangePickerHarness, fixture } = await setupTest();
+
+    await dateRangePickerHarness.selectCalculator(
+      SkyDateRangeCalculatorId.SpecificRange,
+    );
+    fixture.componentInstance.myForm.markAllAsTouched();
```

Comment:
we did so for set value bc consumers would not know that we do not run check for validation unless we blurred. In this scenario im calling markAllAsTouched so that the required errors would throw on the start and end date pickers. So i dont think we should call blur?

Though when a consumer does select a calculator from the dropdown the select does get blurred and causes the required errors to show up anyways....

Reply:
>Though when a consumer does select a calculator from the dropdown the select does get blurred and causes the required errors to show up anyways....

Should we wait until that bug is fixed to before merging this then?

Reply:
the datepicker bug about them being marked as touched when moving to the calendar wouldnt affect this. Honestly I feel more strongly that we shouldn't blur inside the `selectCalculator` method now bc if all a consumer does is switch between two calculators, date range picker does not become blurred. 

in the demo: switch to a calculator with datepickers, the datepickers do not show the required error until the consumer blurs off.